src/full/encoder.rs

```diff
@@ -129,6 +135,39 @@ impl Encoder {
         Ok(full_coded_piece)
     }
```

Comment:
I think we can just call `code_with_buf_coding_vector` from `code_with_coding_vector` instead of replicating the coded-piece compute logic there.